algebra-core/src/serialize/mod.rs

```diff
@@ -49,23 +49,34 @@ pub trait ConstantSerializedSize: CanonicalSerialize {
 ///
 /// If your code depends on `algebra` instead, the example works analogously
 /// when importing `algebra::serialize::*`.
+
 pub trait CanonicalSerialize {
     /// Serializes `self` into `writer`.
     fn serialize<W: Write>(&self, writer: W) -> Result<(), SerializationError>;
 
     fn serialized_size(&self) -> usize;
 
+    /// Serializes `self` into `writer` without compression, and without
+    /// performing validity checks. Should be used *only* when there is no
+    /// danger of adversarial manipulation of the output.
+    #[inline]
+    fn serialize_unchecked<W: Write>(&self, writer: W) -> Result<(), SerializationError> {
+        // TODO make this just serialize
```

Comment:
Is this TODO needed? Where would we just serialize?

Reply:
The comment was extraneous and has been removed